MastodonSDK/Sources/MastodonUI/View/Content/StatusCardControl.swift

```diff
@@ -0,0 +1,337 @@
+        imageView.setContentHuggingPriority(.zero, for: .vertical)
+        imageView.setContentCompressionResistancePriority(.zero, for: .horizontal)
+        imageView.setContentCompressionResistancePriority(.zero, for: .vertical)
+
+        labelStackView.addArrangedSubview(titleLabel)
+        labelStackView.addArrangedSubview(linkLabel)
+        labelStackView.layoutMargins = .init(top: 10, left: 10, bottom: 10, right: 10)
+        labelStackView.isLayoutMarginsRelativeArrangement = true
+        labelStackView.axis = .vertical
+        labelStackView.spacing = 2
+
+        containerStackView.addArrangedSubview(imageView)
+        containerStackView.addArrangedSubview(dividerView)
+        containerStackView.addArrangedSubview(labelStackView)
+        containerStackView.isUserInteractionEnabled = false
+        containerStackView.distribution = .fill
+
+        addSubview(containerStackView)
+        addSubview(highlightView)
+        addSubview(showEmbedButton)
+
+        containerStackView.translatesAutoresizingMaskIntoConstraints = false
+        highlightView.translatesAutoresizingMaskIntoConstraints = false
+        showEmbedButton.translatesAutoresizingMaskIntoConstraints = false
+
+        containerStackView.pinToParent()
+        highlightView.pinToParent()
+        NSLayoutConstraint.activate([
+            showEmbedButton.centerXAnchor.constraint(equalTo: imageView.centerXAnchor),
+            showEmbedButton.centerYAnchor.constraint(equalTo: imageView.centerYAnchor),
+        ])
+
+        addInteraction(UIContextMenuInteraction(delegate: self))
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func configure(card: Card) {
+        if let host = card.url?.host {
+            accessibilityLabel = "\(card.title) \(host)"
+        } else {
+            accessibilityLabel = card.title
+        }
+
+        titleLabel.text = card.title
+        linkLabel.text = card.url?.host
+        imageView.contentMode = .center
+
+        imageView.sd_setImage(
+            with: card.imageURL,
+            placeholderImage: icon(for: card.layout)
+        ) { [weak self] image, _, _, _ in
+            if image != nil {
+                self?.imageView.contentMode = .scaleAspectFill
+            }
+
+            self?.containerStackView.setNeedsLayout()
+            self?.containerStackView.layoutIfNeeded()
+        }
+
+        if let html = card.html, !html.isEmpty {
+            showEmbedButton.isHidden = false
+            self.html = html
+        } else {
+            webView?.removeFromSuperview()
+            webView = nil
+            showEmbedButton.isHidden = true
+            self.html = ""
+        }
+
+        updateConstraints(for: card.layout)
+    }
+
+    public override func didMoveToWindow() {
+        super.didMoveToWindow()
+
+        if let window = window {
+            layer.borderWidth = 1 / window.screen.scale
```

Comment:
Any scenario where we might run into a division-by-zero hero? Not sure if we'd need to check this.